projects/igniteui-angular/src/lib/date-range-picker/date-range-picker.component.ts

```diff
@@ -0,0 +1,875 @@
+        } else {
+            return this.inputFormat || DEFAULT_INPUT_FORMAT;
+        }
+    }
+
+    /** @hidden @internal */
+    public get hasProjectedInputs(): boolean {
+        return this.projectedInputs.length > 0;
+    }
+
+    private get dropdownOverlaySettings(): OverlaySettings {
+        return Object.assign({}, this._dropDownOverlaySettings, this.overlaySettings);
+    }
+
+    private get dialogOverlaySettings(): OverlaySettings {
+        return Object.assign({}, this._dialogOverlaySettings, this.overlaySettings);
+    }
+
+    private get required(): boolean {
+        if (this._ngControl && this._ngControl.control && this._ngControl.control.validator) {
+            const error = this._ngControl.control.validator({} as AbstractControl);
+            return error && error.required;
+        }
+
+        return false;
+    }
+
+    private _value: DateRange;
+    private _ngControl: NgControl;
+    private _statusChanges$: Subscription;
+    private $destroy = new Subject();
+    private _minValue: Date;
+    private _maxValue: Date;
+    private $toggleClickNotifier = new Subject();
+    private _positionSettings: PositionSettings;
+    private _dialogOverlaySettings: OverlaySettings = {
+        closeOnOutsideClick: true,
+        modal: true
+    };
+    private _dropDownOverlaySettings: OverlaySettings = {
+        closeOnOutsideClick: true,
+        modal: false
+    };
+    private onChangeCallback = (...args: any[]) => { };
+    private onTouchCallback = (...args: any[]) => { };
+    private onValidatorChange = (...args: any[]) => { };
+
+    constructor(public element: ElementRef,
+        @Optional() @Inject(DisplayDensityToken) protected _displayDensityOptions: IDisplayDensityOptions,
+        @Inject(LOCALE_ID) private _locale: any,
+        private _injector: Injector) {
+        super(_displayDensityOptions);
+        this.locale = this.locale || this._locale;
+    }
+
+    /**
+     * Opens the date range picker's dropdown or dialog.
+     *
+     * @example
+     * ```html
+     * <igx-date-range-picker #dateRange></igx-date-range-picker>
+     *
+     * <button (click)="dateRange.open()">Open Dialog</button
+     * ```
+     */
+    public open(overlaySettings?: OverlaySettings): void {
+        const range: Date[] = [];
+        if (this.value) {
+            if (this.value.start) {
+                range.push(this.value.start);
+                if (this.value.end) {
+                    range.push(this.value.end);
+                }
+            } else {
+                this.value = { start: null, end: null };
+            }
+        }
+        if (range.length > 0) {
+            this.value = { start: range[0], end: range[range.length - 1] };
+        }
+
+        this.updateCalendar();
+        const settings = this.mode === InteractionMode.Dialog ? this.dialogOverlaySettings : this.dropdownOverlaySettings;
+        if (this.toggleDirective.collapsed) {
```

Comment:
You can now call:
```
this.collapsed
```
Also I would check if collapsed in the first row of the method, and save some unnecessary checks and object assigns.